tests/Unit/Evolution/Systems/Cce/Equations.py

```diff
@@ -0,0 +1,195 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import numpy as np
+
+# Test function for beta equation
+
+
+def compute_integrand_for_beta(dy_j, j, one_minus_y):
+    dy_jbar = np.conj(dy_j)
+    jbar = np.conj(j)
+    dy_j_jbar = j * dy_jbar + jbar * dy_j
+    k_squared = 1.0 + j * jbar
+    return -one_minus_y / 8.0 * (dy_j * dy_jbar \
+                                 - dy_j_jbar**2 / (4.0 * k_squared))
+
+
+# Test functions for Q equation
+
+
+def compute_integrand_for_q_pole_part(eth_beta):
+    return -4.0 * eth_beta
+
+
+def compute_integrand_for_q_regular_part(
+        dy_beta, dy_j, j, eth_dy_beta, eth_j_jbar, eth_jbar_dy_j, ethbar_dy_j,
+        ethbar_j, eth_r_divided_by_r, k, script_aq):
+    eth_dy_jbar = np.conj(ethbar_dy_j)
+    dy_jbar = np.conj(dy_j)
+    ethbar_r_divided_by_r = np.conj(eth_r_divided_by_r)
+    eth_jbar = np.conj(ethbar_j)
+    jbar = np.conj(j)
+    script_aq = - eth_jbar_dy_j / 4.0 + j * eth_dy_jbar / 4.0 \
+                - eth_jbar * dy_j / 4.0 \
+                + eth_j_jbar * (jbar * dy_j + j * dy_jbar)\
+                  / (8.0 * (1.0 + j * jbar)) \
+                + (j * dy_jbar - jbar * dy_j) * eth_r_divided_by_r / 4.0
+    return - (2.0 * script_aq + 2.0 * j * np.conj(script_aq) / k \
+              - 2.0 * eth_dy_beta + ethbar_dy_j / k \
+              - 2.0 * dy_beta * eth_r_divided_by_r \
+              + dy_j * ethbar_r_divided_by_r / k)
+
+
+# Test function for U equation
+
+
+def compute_integrand_for_u(exp_2_beta, j, q, k, r):
+    qbar = np.conj(q)
+    return exp_2_beta / (2.0 * r) * (k * q - j * qbar)
+
+
+# Test functions for W equation
+
+
+def compute_integrand_for_w_pole_part(ethbar_u):
+    eth_ubar = np.conj(ethbar_u)
+    return eth_ubar + ethbar_u
+
+
+def compute_integrand_for_w_regular_part(
+        dy_u, exp_2_beta, j, q, eth_beta, eth_eth_beta, eth_ethbar_beta,
+        eth_ethbar_j, eth_ethbar_j_jbar, eth_j_jbar, ethbar_dy_u,
+        ethbar_ethbar_j, ethbar_j, eth_r_divided_by_r, k, r, script_av):
+    dy_ubar = np.conj(dy_u)
+    ethbar_beta = np.conj(eth_beta)
+    eth_dy_ubar = np.conj(ethbar_dy_u)
+    ethbar_ethbar_beta = np.conj(eth_eth_beta)
+    eth_eth_jbar = np.conj(ethbar_ethbar_j)
+    eth_jbar = np.conj(ethbar_j)
+    ethbar_j_jbar = np.conj(eth_j_jbar)
+    ethbar_r_divided_by_r = np.conj(eth_r_divided_by_r)
+    jbar = np.conj(j)
+    k_squared = (1.0 + j * jbar)
+    qbar = np.conj(q)
+    script_av = eth_beta * eth_jbar + ethbar_ethbar_j / 2.0 \
```

Comment:
You take this as an argument but then compute it

Reply:
It is a temporary buffer, taken by argument to avoid allocation.

Reply:
but this is in python, so does that actually work?

Reply:
oh, oops, I failed to note which file this was commenting on. No, it doesn't have any optimization utility in python. I could name the intermediate step something else, but I still need the argument so that the function signature is the same as the C++, and I still want to compute the intermediate step so that the function is reasonably readable.

Reply:
You can use an underscore `_` for unused arguments, that's the convention